File: scripts/insert-regenerated-models.py
#!/usr/bin/env python3
import json
import re

# Load the parsed models
with open('regenerated-models-parsed.json', 'r') as f:
    models = json.load(f)

# Load the TypeScript file
with open('lib/readwise-data.ts', 'r') as f:
    content = f.read()

# Find the READWISE_MODELS array
match = re.search(r'export const READWISE_MODELS.*?= \[(.*)\];', content, re.DOTALL)
if not match:
    print("❌ Could not find READWISE_MODELS array")
    exit(1)

models_json_str = '[' + match.group(1) + ']'

# Parse the models JSON
try:
    existing_models = json.loads(models_json_str)
    print(f"✅ Parsed {len(existing_models)} existing models")
except json.JSONDecodeError as e:
    print(f"❌ Failed to parse models JSON: {e}")
    exit(1)

# Add the new models to the existing models list
for new_model in models:
    model_obj = {
        "id": f"{new_model['domainSlug']}-{new_model['code'].lower()}",
        "code": new_model['code'],
        "name": new_model['name'],
        "description": new_model['description'],
        "slug": new_model['slug'],
        "domain": new_model['domain'],
        "domainSlug": new_model['domainSlug'],
        "principles": new_model['principles'],
        "examples": [],
        "applications": new_model['applications'],
        "relatedModels": [],
        "sources": [],
        "tags": [],
        "difficulty": "intermediate",
        "createdAt": "2025-10-31T00:00:00.000Z",
        "updatedAt": "2025-10-31T00:00:00.000Z"
    }
    
    existing_models.append(model_obj)
    print(f"📌 Added {new_model['code']}: {new_model['name']}")

# Convert back to JSON
new_models_json = json.dumps(existing_models, indent=2, ensure_ascii=False)

# Replace in the content
new_content = content.replace(
    match.group(0),
    f"export const READWISE_MODELS: MentalModel[] = {new_models_json};"
)

# Write back
with open('lib/readwise-data.ts', 'w') as f:
    f.write(new_content)

print(f"\n✅ Successfully added {len(models)} models!")
print("Total models now:", len(existing_models))

File: scripts/insert-regenerated-models.js
const fs = require('fs');
const path = require('path');

/**
 * Insert the 9 regenerated models into lib/readwise-data.ts
 * at the appropriate domain positions
 */

console.log('📝 Inserting regenerated models into lib/readwise-data.ts...\n');

// Load the parsed models
const parsedPath = path.join(__dirname, '..', 'regenerated-models-parsed.json');
const models = JSON.parse(fs.readFileSync(parsedPath, 'utf8'));

// Load readwise-data.ts
const dataPath = path.join(__dirname, '..', 'lib', 'readwise-data.ts');
let dataContent = fs.readFileSync(dataPath, 'utf8');

// Function to format a model as TypeScript
function formatModel(model) {
  return `  {
    "id": "${model.domainSlug}-${model.code.toLowerCase()}",
    "code": "${model.code}",
    "name": "${model.name}",
    "description": "${model.description}",
    "slug": "${model.slug}",
    "domain": "${model.domain}",
    "domainSlug": "${model.domainSlug}",
    "principles": ${JSON.stringify(model.principles, null, 6).replace(/\n/g, '\n    ')},
    "examples": [],
    "applications": ${JSON.stringify(model.applications, null, 6).replace(/\n/g, '\n    ')},
    "relatedModels": [],
    "sources": [],
    "tags": [],
    "difficulty": "intermediate",
    "createdAt": "${new Date().toISOString()}",
    "updatedAt": "${new Date().toISOString()}"
  }`;
}

// Insert models by domain
let insertedCount = 0;

for (const model of models) {
  console.log(`📌 Inserting ${model.code}: ${model.name}`);
  
  // Find the domain section
  const domainRegex = new RegExp(
    `domain: "${model.domain.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}"[\\s\\S]*?(?=\\n  },\\n  {\\n    "id"|\\n];)`,
    'g'
  );
  
  // Find all models in this domain
  const domainMatches = [...dataContent.matchAll(domainRegex)];
  
  if (domainMatches.length === 0) {
    console.log(`   ❌ Could not find domain: ${model.domain}`);
    continue;
  }
  
  // Get the last match (last model in this domain)
  const lastMatch = domainMatches[domainMatches.length - 1];
  const insertPosition = lastMatch.index + lastMatch[0].length;
  
  // Insert the new model
  const formattedModel = formatModel(model);
  dataContent = dataContent.slice(0, insertPosition) + 
                ',\n' + formattedModel + 
                dataContent.slice(insertPosition);
  
  insertedCount++;
  console.log(`   ✅ Inserted successfully`);
}

// Write the updated content back
fs.writeFileSync(dataPath, dataContent);

console.log(`\n✅ Successfully inserted ${insertedCount} models into lib/readwise-data.ts\n`);
console.log('🔄 Next step: Verify the file compiles and test the website\n');


